### DISCUSSION TITLE
Mathieu is an animal

### DISCUSSION THREAD

**Discussion by luss** [pgsql-io/multicorn2#52] [General]
His ideas are great with simple implementations AND his code looks rock solid.   I'm gonna try and get Jan and/or Affan (who both work for me at pgEdge presently) to forgure out what I broke in the C code (on query predicate pushdown) when I tweaked it to compile in PG13/14.   This will clear up most of the outstanding issues and is a wonderful compliement to the Python work Matyieu is doing to make Multicorn great again.  :-)

**Comment by mfenniak** [comment]
Thanks for the kind words. :-)

Regarding query predicate pushdown, what do we have for specific knowledge about that problem?  I've seen your comments on this, of course, but I don't have a clear sense from the open issues of what problem there is.  The closest is #8 which would require an enhancement to support OR clauses, and the LDAP issues aren't really well defined enough to be reproducible.

The test suite has a good number of tests that use either EXPLAIN or NOTICE logging from execute() to verify that quals are pushed into the FDW.

I'm not saying there isn't a problem... but... I'm not sure that there is one either.

I have discovered some fun crashes when using non-constants though, but since those are unreported I've assumed I'm in novel territory.
